fee-monitor/src/index.ts

```diff
@@ -33,13 +33,42 @@ async function getNextBlockNumber(current: number) {
     }
 }
 
+async function getCurrentTermStartBlockNumber(blockNumber: number): Promise<number | null> {
+    return new Promise((resolve, reject) => {
+        sdk.rpc
+            .sendRpcRequest("chain_getTermMetadata", [blockNumber])
+            .then(result => {
+                if (result === null) {
+                    resolve(null);
+                }
+                const [prevTermLastBlockNumber, currentTermId] = result;
+                if (
+                    typeof prevTermLastBlockNumber === "number" &&
+                    typeof currentTermId === "number"
+                ) {
+                    resolve(prevTermLastBlockNumber + 1);
+                }
+                reject(
+                    Error(
+                        `Expected getTermMetadata to return [number, number] | null but it returned ${result}`,
+                    ),
+                );
+            })
+            .catch(reject);
+    });
+}
+
 async function startFrom() {
     if (process.env.BLOCK_NUMBER) {
         let blockNumber = parseInt(process.env.BLOCK_NUMBER, 10);
         if (isNaN(blockNumber) || blockNumber === 0) {
             throw new Error("BLOCK_NUMBER must be a non-zero positive integer");
         }
-        return blockNumber;
+        const currentTermStart = await getCurrentTermStartBlockNumber(blockNumber);
+        if (currentTermStart === null) {
+            throw new Error("BLOCK_NUMBER is bigger than current best block number");
+        }
+        return currentTermStart;
```

Comment:
They should return `blockNumber` when it is before the first term close. Otherwise, it'll try to scan from scratch.

Reply:
Thank you. I missed that point.